tests/Unit/Domain/CoordinateMaps/Test_Frustum.cpp

```diff
@@ -0,0 +1,171 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include <array>
+#include <catch.hpp>
+#include <random>
+
+#include "Domain/CoordinateMaps/CoordinateMap.hpp"
+#include "Domain/CoordinateMaps/Frustum.hpp"
+//#include "Domain/CoordinateMaps/Wedge3D.hpp"
+#include "Utilities/ConstantExpressions.hpp"
+#include "tests/Unit/Domain/CoordinateMaps/TestMapHelpers.hpp"
+#include "tests/Unit/TestHelpers.hpp"
+
+SPECTRE_TEST_CASE("Unit.Domain.CoordinateMaps.Frustum", "[Domain][Unit]") {
+  const std::array<double, 3> lower_corner{{-1.0, -1.0, -1.0}};
+  const std::array<double, 3> upper_corner{{1.0, 1.0, 1.0}};
+  const std::array<double, 3> test_point1{{-0.1, 0.3, 0.1}};
+  const std::array<double, 3> test_point2{{0.5, 0.7, -0.5}};
+  const std::array<double, 3> test_point3{{0.9, -1.0, 0.4}};
+  const std::array<std::array<double, 2>, 4> face_vertices{
+      {{{0.0, 0.0}}, {{2.0, 4.0}}, {{-1.0, -1.0}}, {{3.0, 5.0}}}};
+  const double lower_bound = 1.0;
+  const double upper_bound = 2.0;
+
+  for (const auto& direction : Direction<3>::all_directions()) {
```

Comment:
See if you can factor this into a function that can be used in other tests
Loop over vector of test points
You could pass random test numbers to the factored out function, then remove the last test in this file